tests/unit/test_download.py

```diff
@@ -157,6 +157,43 @@ def test_unpack_http_url_bad_downloaded_checksum(mock_unpack_file):
         rmtree(download_dir)
 
 
+def test_download_http_url_with_directory_traversal(data):
+    """
+    It should download file to temp_sub_dir
```

Comment:
I would be more explicit in the docstring and say something like, "Test downloading a file when the content-disposition header contains a filename with a ".." path part."